File: frontend/src/api/model/DocumentStatusResponse.ts
import { ProcessingStatus } from "./ProcessingStatus";

/**
 * Response DTO for document status inquiry
 */
export interface DocumentStatusResponse {
  documentId: string;
  filename: string;
  status: ProcessingStatus;
  errorMessage: string | null;
  totalParagraphs: number;
  embeddedParagraphs: number;
  embeddingComplete: boolean;
  uploadedAt: string;
  processedAt: string | null;
}

File: backend/src/main/java/com/unfinitas/backend/api/dto/DocumentStatusResponse.java
package com.unfinitas.backend.api.dto;

import com.unfinitas.backend.core.ingestion.model.MoeDocument;
import com.unfinitas.backend.core.ingestion.model.ProcessingStatus;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Response DTO for document status inquiry
 */
public record DocumentStatusResponse(
        UUID documentId,
        String fileName,
        Long fileSize,
        Integer pageCount,
        Integer paragraphCount,
        ProcessingStatus status,
        String errorMessage,
        LocalDateTime createdAt
) {
    /**
     * Create response from MoeDocument entity with paragraph count
     */
    public static DocumentStatusResponse from(MoeDocument document, int paragraphCount) {
        return new DocumentStatusResponse(
                document.getId(),
                document.getFileName(),
                document.getFileSize(),
                document.getPageCount(),
                paragraphCount,
                document.getProcessingStatus(),
                document.getErrorMessage(),
                document.getCreatedAt()
        );
    }
}
